### PR TITLE
dts: arm: st: new compatible, for instance st,stm32g0-exti

### PR DESCRIPTION
**Pull request by Desvauxm-st**
by-passed thanks to a new compatible
for instance st,stm32g0-exti added to the matching targets: G0/U5/L5/MP1:

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Please fix commit title.

Also please split into commits:
- dts
- driver changes

File: dts/arm/st/l5/stm32l5.dtsi [hunk 1]
```diff
@@ -134,7 +134,7 @@
 		};
 
 		exti: interrupt-controller@4000f400 {
-			compatible = "st,stm32-exti";
+			compatible = "st,stm32g0-exti", "st,stm32-exti";
```
[the remark this answers is not in the document]
Comment (Desvauxm-st) [reply dts/arm/st/l5/stm32l5.dtsi:137]: I don't know how to do that

[the remark this answers is not in the document]
Comment (Desvauxm-st) [reply dts/arm/st/l5/stm32l5.dtsi:158]: add the new  file st,stm32g0-exti.yaml into dts/bindings/interrupt-controller

**Comment by erwango** [comment]
Please rebase on top of latest main branch and remove "merge commit".

**Comment by Desvauxm-st** [comment]
see #55844

**Pull request closed without merging**